erasure-utils.go

```diff
@@ -44,21 +44,22 @@ func newHash(algo string) hash.Hash {
 	}
 }
 
+// hashSum calculates the hash of the entire path and returns.
 func hashSum(disk StorageAPI, volume, path string, writer hash.Hash) ([]byte, error) {
 	startOffset := int64(0)
-	// Read until io.EOF.
+	// Read until io.EOF and write the read buffer to hash writer.
 	for {
-		buf := make([]byte, blockSizeV1)
-		n, err := disk.ReadFile(volume, path, startOffset, buf)
+		buf, err := disk.ReadFile(volume, path, startOffset, blockSizeV1)
```

Comment:
to avoid extra ReadFile() call (which gets io.EOF) to the disk this can be done:
```
func hashSum(disk StorageAPI, volume, path string, writer hash.Hash) ([]byte, error) { 
        startOffset := int64(0)                                                        
        // Read until io.EOF and write the read buffer to hash writer.                 
        for {                                                                          
                buf, err := disk.ReadFile(volume, path, startOffset, blockSizeV1)      
                if len(buf) > 0 {                                                      
                        writer.Write(buf)                                              
                        startOffset += int64(len(buf))                                 
                }                                                                      
                if err == io.EOF || err == io.ErrUnexpectedEOF {                       
                        break                                                          
                }                                                                      
                if err != nil {                                                        
                        return nil, err                                                
                }                                                                      
        }                                                                              
        // Return the final hash sum.                                                  
        return writer.Sum(nil), nil                                                    
}  
```